tests/utils.py

```diff
@@ -1,22 +1,45 @@
 import contextlib
 import os
+import sys
+import platform
 import subprocess
+
+# Hack for python 3.6
+subprocess._cleanup = lambda: None
```

Comment:
If this is only for Python 3.6, could we add a check here for that version?

Something like:

```suggestion
if (3, 6) <= sys.version_info < (3, 7):
    subprocess._cleanup = lambda: None
```

Reply:
Turns out I don't think this was an bug with python 3.6 it was me being lazy else where, I read an bug report and got excited thinking I'd solved it